spec-tests/src/test/java/org/candlepin/spec/EntitlementCertificateV3SpecTest.java

```diff
@@ -17,121 +17,493 @@
+        if (CandlepinMode.isHosted()) {
+            hostedTestApi.createOwner(owner);
+            product = hostedTestApi.createProduct(product);
+            product30 = hostedTestApi.createProduct(product30);
+            content = hostedTestApi.createContent(content);
+            archContent = hostedTestApi.createContent(archContent);
+            hostedTestApi.addContentToProduct(product.getId(), Map.of(content.getId(), false));
+            hostedTestApi.addContentToProduct(product.getId(), Map.of(archContent.getId(), false));
+            subscription = Subscriptions.random(owner, product)
+                .contractNumber("12345")
+                .accountNumber("6789")
+                .orderNumber("order1");
+            subscription = hostedTestApi.createSubscription(subscription);
+            subscription30 = Subscriptions.random(owner, product30)
+                .contractNumber("123456")
+                .accountNumber("67890")
+                .orderNumber("order2");
+            subscription30 = hostedTestApi.createSubscription(subscription30);
+            AsyncJobStatusDTO refresh = ownerApi.refreshPools(owner.getKey(), false);
+            if (refresh != null) {
+                client.jobs().waitForJob(refresh.getId());
+            }
+        }
+        else {
+            product = ownerProductApi.createProductByOwner(owner.getKey(), product);
+            product30 = ownerProductApi.createProductByOwner(owner.getKey(), product30);
+            content = ownerContentApi.createContent(owner.getKey(), content);
+            archContent = ownerContentApi.createContent(owner.getKey(), archContent);
+            product = ownerProductApi.addContent(owner.getKey(), product.getId(), content.getId(), false);
+            product = ownerProductApi.addContent(owner.getKey(), product.getId(), archContent.getId(), false);
+            pool = Pools.random(product)
+                .contractNumber("12345")
+                .accountNumber("6789")
+                .orderNumber("order1");
+            pool = ownerApi.createPool(owner.getKey(), pool);
+            pool30 = Pools.random(product30)
+                .contractNumber("123456")
+                .accountNumber("67890")
+                .orderNumber("order2");
+            pool30 = ownerApi.createPool(owner.getKey(), pool30);
+        }
+        system = Consumers.random(owner)
+            .name(StringUtil.random("system"))
+            .type(new ConsumerTypeDTO().label("system"))
+            .facts(Map.of("system.certificate_version", "3.4", "uname.machine", "i386"));
+        system = consumerApi.register(system);
+    }
+
+    /**
+     * This test covers the case where the system supports 3.0 certs, but
+     * the server is creating 3.4 certs, and the product contains attributes
+     * supported by 3.0.
+     * @throws Exception
+     */
+    @Test
+    @DisplayName("generated a version 3.4 certificate when requesting a 3.0 certificate")
```

Comment:
Let's follow the convention to start with `should/should not` for all test method/displaynames in the file like we've been doing elsewhere